runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaDoFnRunners.java

```diff
@@ -369,6 +419,7 @@ public void finishBundle() {
         // RemoteBundle close blocks until all results are received
         remoteBundle.close();
         emitResults();
+        emitMetrics();
```

Comment:
since we included all the other state ops in the metrics, let's emitMetrics() after clear the state. That will cover the full operation of invoking the bundle.

Reply:
metric calculation relies on state count, therefore, we cannot move it after clear the state. 

on the other hand, if we preserve the count before clearing, we end up reading state count when we do not need emit metrics.

as agreed offline, we will keep it as it is for now.